doc/src/specifications/app-architecture/plugins.md

```diff
@@ -112,27 +112,38 @@ Blockchains typically only have public data. If private client-side data is coll
+### Host-dependent data backing
 
-This `clientdata` plugin provides a high level key/value API for plugins.
+It is up to the host platform to determine how to implement the key-value interface. For example, in the current implementation of the host:plugin for the web platform, the key-value implementation uses LocalStorage as its data backing, because it is a synchronous interface which is easiest to integrate with wasm.
 
-```
-    get: func(key: string) -> option<list<u8>>;
-    set: func(key: string, value: list<u8>) -> result<_, error>;
-    delete: func(key: string);
-```
+Note: This imposes a major restriction on the amount of local data storage available for psibase plugins: Currently there is [a hard limit](https://developer.mozilla.org/en-US/docs/Web/API/Storage_API/Storage_quotas_and_eviction_criteria#web_storage) for the total amount of data stored in LocalStorage across all psibase apps in a given browser, since storage across all apps is managed by a single domain. This restriction will be lifted in a future implementation of plugin storage that uses a different data backing layer such as IndexedDB.
 
-Every plugin has its own independent key space, and can only directly read/write its own data. Data may be shared between plugins only if explicilty exposed through the plugin's API.
+### API
 
-### Data backing
+The current API for key-value operations allows one to select one of multiple different databases for reading/writing. There are two database modes: Non-Transactional and Transactional.
 
-The `wasi-keyvalue` implementation currently uses LocalStorage as its data backing, because it is a synchronous interface which is easiest to integrate with wasm.
+- Non-Transactional - A Non-Transactional database is independent of the currently running transaction, and any writes will be persisted regardless of the success/failure of the current plugin function or any consequent server-side actions.
+- Transactional - Writes to the Transactional database are dependent on the success of the currently running transaction. If an action fails (or a consequent server-side transaction), then any writes to the Transactional database will be rolled back.
 
-Note: This imposes a major restriction on the amount of local data storage available for psibase plugins: Currently there is [a hard limit](https://developer.mozilla.org/en-US/docs/Web/API/Storage_API/Storage_quotas_and_eviction_criteria#web_storage) for the total amount of data stored in LocalStorage across all psibase apps in a given browser, since storage across all apps is managed by a single domain. This restriction will be lifted in a future implementation of plugin storage that uses a different data backing layer such as IndexedDB.
+There are three different settings for data-duration that may be used:
+
+- Ephemeral - An ephemeral database is wiped at the start of every new execution context. This data never persists longer than the currently executing callstack.
```

Comment:
I think the exact point at which it is wiped is an implementation detail. In fact, I prefer to think of it as every execution context getting a completely independent database. It only needs to be wiped because it happens to reuse the same storage as a previous instance.